backend/src/index.ts

```diff
@@ -72,39 +77,75 @@ const processSourceBlocks = (target: Target) => async (source: Source) => {
     const targetApi = await createApi(config.targetChainUrl);
 
     const target = new Target({ api: targetApi, logger, state });
+    const master = getAccount(config.accountSeed);
 
-    const sources = await Promise.all(
-      config.sourceChains.map(async ({ url, parachains }) => {
-        const api = await createApi(url);
-        const chain = await api.rpc.system.chain();
-        const sourceSigner = getAccount(`${config.accountSeed}/${chain}`);
-        const paraSigners = parachains.map(({ paraId }) => getAccount(`${config.accountSeed}/${paraId}`));
-
-        // TODO: can be optimized by sending batch of txs
-        // TODO: master has to delegate spending to sourceSigner and paraSigners
-        const master = getAccount(config.accountSeed);
-        for (const delegate of [sourceSigner, ...paraSigners]) {
-          // send 1.5 units
-          await target.sendBalanceTx(master, delegate, 1.5);
+    if (args.length && (args[0] === 'archive')) {
+      if (!config.archives) {
+        throw new Error("Archives are not provided");
+      }
+
+      const archives = await Promise.all(config.archives.map(async ({ path, url }) => {
+        if (!path) {
+          throw new Error("Archive path is not provided");
         }
 
-        // check if feed already exists
-        const feedId = await target.getFeedId(sourceSigner);
-        const parachainsMap = await createParachainsMap(target, parachains, paraSigners);
+        const api = await createApi(url);
+        const chain = (await api.rpc.system.chain()).toString() as ChainName;
+        const signer = getAccount(`${config.accountSeed}/${chain}`);
+        await target.sendBalanceTx(master, signer, 1.5);
+        const feedId = await target.getFeedId(signer);
 
-        return new Source({
+        return new ChainArchive({
           api,
-          chain: chain.toString() as ChainName,
-          parachainsMap,
-          logger,
+          path,
+          chain,
           feedId,
-          signer: sourceSigner,
+          logger,
+          signer,
           state,
         });
-      })
-    );
+      }))
 
-    sources.forEach(processSourceBlocks(target));
+      archives.forEach(async archive => {
+        let nonce = (await target.api.rpc.system.accountNextIndex((archive.signer as KeyringPair).address)).toBn();
+        for await (const blockData of archive.getBlocks()) {
+          target.sendBlockTx(blockData, nonce);
+          nonce = nonce.add(new BN(1));
+        }
+      });
+    } else {
+      // default - processing blocks from RPC API
+      const sources = await Promise.all(
+        config.sourceChains.map(async ({ url, parachains }) => {
+          const api = await createApi(url);
+          const chain = await api.rpc.system.chain();
```

Comment:
This would bring typing closer to the source of data:
```suggestion
          const chain = (await api.rpc.system.chain()).toString() as ChainName;
```